src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -97,19 +79,29 @@ private void AddNullableAnnotations(TypeSymbolWithAnnotations typeOpt)
                 return;
             }
 
+            if (ShouldPrintNullableAnnotation(typeOpt))
+            {
+                AddPunctuation(typeOpt.NullableAnnotation.IsAnyNullable() ? SyntaxKind.QuestionToken : SyntaxKind.ExclamationToken);
+            }
+        }
+
+        private bool ShouldPrintNullableAnnotation(TypeSymbolWithAnnotations typeOpt)
+        {
             if (format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier) &&
                 !typeOpt.IsNullableType() && !typeOpt.IsValueType &&
```

Comment:
>typeOpt [](start = 17, length = 7)

Looks like `typeOpt` may be null (actually `IsNull`) when called from `VisitArrayType` with an array type from VB.